.github/workflows/publish.yml

```diff
@@ -34,9 +34,26 @@ jobs:
         shell: bash
         run: |
           cd web3rpc/codegen && gradle wrapper && gradle wrapper clean :deployJar && cd - &&\
-          cd web3rpc/rpc-specs && npm install && bash -x generate-namespace.sh && cd -
+          cd web3rpc/rpc-specs && npm install && bash -x generate-namespace.sh && cd 
+
+      - name: Verify Changed files
+        uses: tj-actions/changed-files@v39
+        id: verify-changed-files
+        with:
+          files_yaml: | 
+            ethers-ext:  
+              - "ethers-ext/package.json"
+            web3js-ext:  
+              - "web3js-ext/package.json"
+            web3j-ext:
+              - "web3j-ext/web3j-ext/build.gradle"
+            web3py-ext:
+              - "web3py-ext/setup.py"
+            web3rpc:
+              - "web3rpc/rpc-specs/**"
 
       - name: generate python sdk
+        if: steps.verify-changed-files.outputs.web3py-ext_any_changed == 'true'     
```

Comment:
I think `web3rpc` needs to be build before publish `web3py-ext`.
`web3rpc/sdk/client/python/python-generate.sh` will create `*api.py` files and then copy these to the `web3py-ext/web3py_ext/web3rpc`